File: 45-jump-game-ii/jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int i=0;
        int cstp=0;
        int mxstp=0;
        int c=0;
        while(i<nums.size()-1){
            mxstp=max(mxstp,nums[i]+i);
            if(cstp==i){
                c+=1;
                cstp=mxstp;
            }
            i+=1;
        }
        return c;
    }
};

File: 45-jump-game-ii/jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        c=0
        cs=0
        ms=0
        i=0
        while i<len(nums)-1:
            ms=max(ms,nums[i]+i)
            if cs==i:
                c+=1
                
                cs=ms
            i+=1
        return c